### PR TITLE
refactor: update `blas/ext/base/scusumkbn2` to follow current project conventions

### LINKED ISSUE #1518: [RFC]: refactor `blas/ext/base/scusumkbn2` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/scusumkbn2`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/scusumkbn2) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by headlessNode**
Resolves #1518 

## Description

> What is the purpose of this pull request?

This pull request:

-   refactors `blas/ext/base/scusumkbn2` to follow current project conventions

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1518 
-   progresses #1152

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/binding.gyp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/binding.gyp]: We don't update the copyright year of an existing file per our conventions.

```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/benchmark/c/benchmark.length.c [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/benchmark/c/benchmark.length.c]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/benchmark/c/Makefile [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #/
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/benchmark/c/Makefile]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/benchmark/benchmark.ndarray.native.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/benchmark/benchmark.ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/benchmark/benchmark.ndarray.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/benchmark/benchmark.ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/benchmark/benchmark.native.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/benchmark/benchmark.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/benchmark/benchmark.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/benchmark/benchmark.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/README.md [hunk 8]
```diff
@@ -2,7 +2,7 @@
 
 @license Apache-2.0
 
-Copyright (c) 2020 The Stdlib Authors.
+Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/README.md]:
```suggestion
Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/src/Makefile [hunk 9]
```diff
@@ -1,7 +1,7 @@
 #/
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/src/Makefile]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/test/test.ndarray.native.js [hunk 10]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/test/test.ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/test/test.scusumkbn2.js [hunk 11]
```diff
@@ -328,7 +327,7 @@ tape( 'the function supports view offsets', function test( t ) {
 	x1 = new Float32Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // begin at 2nd element
 	y1 = new Float32Array( y0.buffer, y0.BYTES_PER_ELEMENT*3 ); // begin at the 4th element
 
-	N = floor( x0.length / 2 );
+	N = 3;
 
 	scusumkbn2( N, 0.0, x1, -2, y1, 1 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/test/test.scusumkbn2.js:332]:
```suggestion
	N = 3;

	scusumkbn2( 3, 0.0, x1, -2, y1, 1 );
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/test/test.scusumkbn2.native.js [hunk 12]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/test/test.scusumkbn2.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/test/test.scusumkbn2.js [hunk 13]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/test/test.scusumkbn2.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/examples/index.js [hunk 14]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/examples/index.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/examples/c/example.c [hunk 15]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/examples/c/example.c]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/docs/types/index.d.ts [hunk 16]
```diff
@@ -1,7 +1,7 @@
 /*
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/docs/types/index.d.ts]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/docs/types/test.ts [hunk 17]
```diff
@@ -1,7 +1,7 @@
 /*
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/docs/types/test.ts]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/examples/c/Makefile [hunk 18]
```diff
@@ -1,7 +1,7 @@
 #/
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/examples/c/Makefile]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/include.gypi [hunk 19]
```diff
@@ -1,6 +1,6 @@
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/include.gypi]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/lib/index.js [hunk 20]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/lib/index.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/include/stdlib/blas/ext/base/scusumkbn2.h [hunk 21]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/include/stdlib/blas/ext/base/scusumkbn2.h]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/src/scusumkbn2.c [hunk 22]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/src/scusumkbn2.c]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/test/test.js [hunk 23]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/test/test.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/test/test.ndarray.js [hunk 24]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/test/test.ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/lib/scusumkbn2.native.js [hunk 25]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/lib/scusumkbn2.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/lib/ndarray.native.js [hunk 26]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/lib/ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/lib/scusumkbn2.js [hunk 27]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/lib/scusumkbn2.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/lib/native.js [hunk 28]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/lib/native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/lib/main.js [hunk 29]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/lib/main.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/lib/ndarray.js [hunk 30]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/lib/ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

**Review by Planeshifter — CHANGES_REQUESTED** [review]
This looks overall very good. Thank you for working on it!

Notice I had to revert the copyright year changes: It is our convention to not update them when files are edited, only when they are first committed to the repository.

One thing: Please edit the `manifest.json` file to be 2-space indented and to have `benchmark` and `example` configurations.

**Comment by headlessNode** [comment]
@Planeshifter could you please elaborate a bit about adding `benchmark` and `example` configurations to `manifest.json`? 
I've looked at various merged PRs for the similar issues but I'm unsure of how to implement. Please guide me in this mattter.

**Comment by headlessNode** [comment]
@Planeshifter I've added the suggested changes, please check.

**Review by Planeshifter — APPROVED** [review]
Thanks! Checks out, looks all good to me now after fixing one more copy-paste error. Will merge shortly.

File: lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/src/addon.c [hunk 31]
```diff
@@ -0,0 +1,47 @@
...
+*
+* @private
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 6 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_FLOAT( env, alpha, argv, 1 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumkbn2/src/addon.c]:
```suggestion
	STDLIB_NAPI_ARGV_FLOAT( env, sum, argv, 1 );
```

**Pull request merged**